secio/src/stream_cipher.rs

```diff
@@ -0,0 +1,109 @@
+
+use aes_ctr::stream_cipher::generic_array::GenericArray;
+use aes_ctr::stream_cipher::{NewFixStreamCipher, StreamCipherCore};
+use aes_ctr::{Aes128Ctr, Aes256Ctr};
+
+#[derive(Clone, Copy)]
+pub enum KeySize {
+    KeySize128,
+    KeySize256,
+}
+
+/// Returns your stream cipher depending on `KeySize`.
+#[cfg(not(all(feature = "aes-all", any(target_arch = "x86_64", target_arch = "x86"))))]
+pub(crate) fn ctr(key_size: KeySize, key: &[u8], iv: &[u8]) -> Box<StreamCipherCore + 'static> {
+    ctr_int(key_size, key, iv)
+}
+ 
+/// Returns your stream cipher depending on `KeySize`.
+#[cfg(all(feature = "aes-all", any(target_arch = "x86_64", target_arch = "x86")))]
+pub(crate) fn ctr(key_size: KeySize, key: &[u8], iv: &[u8]) -> Box<StreamCipherCore + 'static> {
+    if *aes_alt::AES_NI {
```

Comment:
I haven't really digged into the crate, but from what I understand the `aes_ctr` crate already does that, no?
It would **really really really** be great if we didn't have platform-specific code.

Reply:
aes_ctr run compile time switch between NI and soft implementation (to activate NI only crate flags from original PR msg should be use). A runtime check is not in aes_ctr crate, that is what eas-all feature does, but honestly I doubt that there is a lot of x86 cpu that could not run NI (I did my tests with qemu).
The best thing to do would be to try to PR runtime NI check to RustCrypto (not directly this code as dynamic type usage does not fit RustCrypto crates design), not sure if it would be easily accepted.
Another option would be to simply remove the 'aes-all' feature as I do not know if it is wanted (usecase would be single x86 build supporting old x86 cpu, it seems reasonable to me to defaults to NI activated build (binary distribution) and ask users to build from source if they are using x86 cpu that does not support aes NI and sse3). For info in #371 Sergei low end config could run aes-ni.